include/ibamr/FEMechanicsBase.h

```diff
@@ -315,13 +315,40 @@ class FEMechanicsBase : public SAMRAI::tbox::Serializable
      * @note       It is @em NOT possible to register multiple surface force
      *             functions with this class.
      */
-    void registerLagSurfaceForceFunction(const LagSurfaceForceFcnData& data, unsigned int part = 0);
+    virtual void registerLagSurfaceForceFunction(const LagSurfaceForceFcnData& data, unsigned int part = 0);
 
     /*!
      * Get the Lagrangian surface force function data.
      */
     LagSurfaceForceFcnData getLagSurfaceForceFunction(unsigned int part = 0) const;
 
+    /*!
+     * Function signature for specifying the energy functional that determines
+     * the pressure.
+     */
+    using VolumetricEnergyDerivativeFcn = std::function<double(double)>;
```

Comment:
While I do prefer `std::function` to raw function pointers I think it would be better to use a consistent style (i.e., use a function pointer here).

Reply:
Boo. OK.

Reply:
We should convert all of these at some point but I think we are just going to confuse people if they see both (sorry)

Reply:
Yeah you are right.

Can we add to the list of starter projects?